openedx/features/enterprise_support/api.py

```diff
@@ -311,34 +355,40 @@ def enterprise_customer_for_request(request):
             # Check if there's an Enterprise Customer such that the linked SSO provider
             # has an ID equal to the ID we got from the running pipeline or from the
             # request tpa_hint URL parameter.
-            ec_uuid = EnterpriseCustomer.objects.get(
+            enterprise_customer_uuid = EnterpriseCustomer.objects.get(
```

Comment:
If we have already retrieved the `EnterpriseCustomer` from the DB here, why do it again using the API down below?

Reply:
Looking at this code it seems we are trying to get `enterprise_customer_uuid` from 3 different ways:
1. by getting SSO provider related enterprise customer. We have `sso_provider_id` from either using url query parameter `tpa_hint` or using third-party auth pipeline.
2. by getting enterprise customer either against the url query parameter `enterprise_customer` or from enterprise cookie (accessing from Ecom)
3. by accessing enterprise API `enterprise-learner` and checking if the requesting user is linked to an enterprise 

We are doing all this because this method is being used at different places. @haikuginger can provide more insight as he is the author.
I have just updated the variable names prefix from `ec_` to `enterprise_coustomer_` for better readability.

Reply:
@zubair-arbi I'm suggesting that we should clean this function up while we are in here. It doesn't make any sense for us to make an API call to get something that we already have.

Reply:
@douglashall @haikuginger  I have removed the call the enterprise learner API in the method `enterprise_customer_for_request` and also updated the related test. Please have another look.